ibis/pandas/execution.py

```diff
@@ -135,6 +138,34 @@ def execute_cast_series_date(op, data, type, scope=None):
 }
 
 
+@execute_node.register(ops.UnaryOp, pd.Series)
+def execute_series_unary_op(op, data, scope=None):
+    function = getattr(np, type(op).__name__.lower())
+    if data.dtype == np.dtype(np.object_):
+        return data.apply(functools.partial(execute_node, op, scope=scope))
+    return function(data)
+
+
+@execute_node.register(
+    ops.Log, pd.Series, (pd.Series, numbers.Real, decimal.Decimal, type(None))
+)
+def execute_series_log_with_base(op, data, base, scope=None):
+    if data.dtype == np.dtype(np.object_):
+        func = np.vectorize(functools.partial(execute_node, op, scope=scope))
+        return pd.Series(func(data, base), index=data.index, name=data.name)
```

Comment:
Perhaps this bit could be factored out into a helper function since it's repeated a couple times (in case it's useful in future execution rules)